config/StoredProcedures/AddLogViewMapEntry.sql

```diff
@@ -0,0 +1,73 @@
+DROP PROCEDURE IF EXISTS AddLogViewMapEntry;
+
+DELIMITER $$
+CREATE PROCEDURE AddLogViewMapEntry
+(IN _Username varchar(25), IN _LogID int unsigned)
+BEGIN
+
+DECLARE test int;
+DECLARE message varchar(100);
+DECLARE flag boolean;
+
+SET message = 'Error:';
+SET flag = false;
+
+# check if username exists
+SELECT EXISTS(
+	SELECT 1
+	FROM Users
+	WHERE Username = _Username)
+INTO test;
+
+IF test = 0 THEN
+	SET flag = true;
+	SET message = CONCAT(message, ' Username: ', _Username, ' doesn\'t exist. ');
+END IF;
+
+SET test = 0;
+
+# check if logId exists
+SELECT EXISTS(
+	SELECT 1
+	FROM Logs
+	WHERE LogID = _LogID)
+INTO test;
+
+IF test = 0 THEN
+	SET flag = true;
+	SET message = CONCAT(message, ' LogID: ', _LogID, ' doesn\'t exist.');
+END IF;
+
+SET test = 0;
+
+IF flag = false THEN # no foreign key errors
+	# check if entry already exists
+	SELECT EXISTS(
+		SELECT 1
+		FROM LogViewMap
+		WHERE Username = _Username AND LogID = _LogID)
+	INTO test;
+
+	IF test = 1 THEN
+		# entry already exists
+		# show existing entry
+		SELECT *
```

Comment:
Can we also maybe try an IF NOT EXISTS?
IF NOT EXISTS(SELECT * FROM LogViewMap WHERE LogID = _LogId AND Username = _Username)
INSERT INTO LogViewMap (LogID, Username)
VALUES (_LogID, _Username)